filament/src/ShadowMap.cpp

```diff
@@ -780,6 +780,7 @@ size_t ShadowMap::intersectFrustumWithBox(
 
     // at this point if we have 8 vertices, we can skip the rest
     if (vertexCount < 8) {
+        Frustum frustum(wsFrustumCorners);
```

Comment:
Just a comment here (don't change it for now). I think it's a little strange to have to recompute the frustum from the corners, because we started from coordinates in clip-space. And it should be trivial to go from that to a Frustum in light space.
 The frustum planes will be:
left {-1,0,0,-1}
right {1,0,0,1}
bottom {0,-1,0,-1}
top {0,1,0,1}
near {0,0,1,near}
far {0,0,-1,far}

then you can just transform them by the transpose(projection()) and that will give you the planes in light space. And as you can see, there are so many zeros that the transform is actually trivial:
```
            float4 p = transpose(projection} * float4{ -1, 0, 0, x };
            plane = float4{ normalize(p.xyz), 0 };
```

which will be more efficient and accurate than doing all the `cross()`

Reply:
Why does the plane have a 0 for the w component? Wouldn't that affect the "position" of the plane?

Reply:
I guess to rephrase this, wouldn't the near and far planes need special treatment? (because their w component cannot be 0, as they don't cross the origin)

Reply:
yes they would, I typed too fast